examples/sealed-auctions/src/types.rs

```diff
@@ -0,0 +1,83 @@
+extern crate chrono;
+
+use std::{collections::HashSet, vec::Vec}; // `use` statements can go anywhere
+
+pub type UserId = mantle::types::Address;
+pub type ItemId = usize;
+
+// A bid struct with the bidder and latest bid
+#[derive(Serialize, Deserialize, Clone, Debug, Eq)]
+pub struct Bid {
+    pub bidder: UserId,
+    pub value: usize,
+}
+
+impl PartialEq for Bid {
+    fn eq(&self, other: &Bid) -> bool {
+        self.bidder == other.bidder
+    }
+}
+
+impl PartialOrd for Bid {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        self.value.partial_cmp(&other.value)
+    }
+}
+
+impl std::hash::Hash for Bid {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.bidder.hash(state);
+    }
+}
+
+impl std::borrow::Borrow<UserId> for Bid {
+    fn borrow(&self) -> &UserId {
+        &self.bidder
+    }
+}
+
+// The auction struct for every auction that has taken place on the platform
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+pub struct Auction {
+    pub item_id: ItemId,
+    pub bids: HashSet<Bid>,
+    pub reserve: usize,
+    pub max_bid: usize,
+    pub realized: usize,
```

Comment:
I'd use explicitly sized types here. `usize` is 32 bits on Wasm but 64 on x86_64 (read: during testing). This could (emphasis on could) maybe (emphasis on maybe) lead to pernicious bugs in production that aren't caught in tests.

Reply:
Should we disallow the use of usize, or provide a warning then?

Reply:
The issue is that disabling it is unintuitive to disable it. I wonder if I can just change it to u64 since that'll never fail.

Reply:
ok, so the good news is that usize is disabled. the bad news is that the error span doesn't point to the right place in a type that uses usize.

Reply:
tl;dr: usize won't get past idl-gen

Reply:
Changed all `usize` to `u64`. Thanks for the catch and color.